worktree.go

```diff
@@ -897,7 +897,7 @@ func (w *Worktree) addIndexFromFile(name string, h plumbing.Hash, idx *indexBuil
 		Name:       name,
 		Mode:       mode,
 		ModifiedAt: fi.ModTime(),
-		Size:       uint32(fi.Size()),
+		Size:       uint32(fi.Size()), //nolint:gosec // G115: file size fits in uint32 for git index
```

Comment:
> file size fits in uint32 for git index

I think this is not correct when considering very large files. IIRC upstream handles this by using bitmask and setting an arbitrary value for a size it knows to be rather large.

Instead of ignoring the lint error, we are better off checking `fi.Size()` and erroring if greater than `uint32`. Supporting larger files can be a follow-up issue/PR.